### PR TITLE
Host warning tan (EXPOSUREAPP-8953)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Tan UI screen and integration with new submission flow + duration fix for previous screen.

Screenshot 2021-08-23 at 16 41 10 Screenshot 2021-08-23 at 16 50 27

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/duration/TraceLocationWarnDurationViewModel.kt [hunk 1]
```diff
@@ -86,6 +92,26 @@ class TraceLocationWarnDurationViewModel @AssistedInject constructor(
...
     ) {
         fun formattedDateTime() = "${localDateTime.toDayFormat()} ${localDateTime.toShortTimeFormat()}"
+
+        fun getReadableDuration(resources: Resources): String {
+            return duration.toReadableDuration(
+                suffix = resources.getString(R.string.tracelocation_organizer_duration_suffix)
+            )
+        }
+
+        fun getTraceLocationWarnDuration(traceLocation: TraceLocation) = TraceLocationWarnDuration(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/duration/TraceLocationWarnDurationViewModel.kt:102]: maybe `OrganizerCheckIn` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/tan/TraceLocationWarnTanFragment.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.state.observe2(this) {
+            binding.apply {
+                uiState = it
+                tanContent.submissionTanCharacterError.setGone(it.areCharactersCorrect)
+                if (it.isCorrectLength) {
+                    tanContent.submissionTanError.setGone(it.isTanValid)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/tan/TraceLocationWarnTanFragment.kt:44]: there is `isGone` from `core-ktx`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/tan/TraceLocationWarnTanFragment.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                requireActivity(),
+                R.string.trace_location_organiser_warn_tan_error_header,
+                it.errorMessage.get(requireContext()),
+                R.string.trace_location_organiser_warn_tan_error_ok_button,
+                null,
+                true,
+                ::goBack
+            )
+            DialogHelper.showDialog(errorDialog)
+        }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/tan/TraceLocationWarnTanFragment.kt:90]: you should use ` it.toErrorDialogBuilder(requireContext()).show()` to display the right message

File: Corona-Warn-App/src/main/res/values/event_registration_strings.xml [hunk 4]
```diff
@@ -297,6 +297,13 @@
     <!-- XTXT: Warn Duration Screen: duration hint -->
     <string name="trace_location_organiser_warn_duration_duration_hint">"Dauer"</string>
 
+    <!--  Warn Tan Screen -->
+    <!-- XHED: Warn Tan Screen: header -->
+    <string name="trace_location_organiser_warn_tan_header">"In Vertretung warnen"</string>
+    <!-- XHED: Warn Tan Screen: error header -->
+    <string name="trace_location_organiser_warn_tan_error_header">"Fehler"</string>
+    <!-- XBUT: Warn Tan Screen: error ok button -->
+    <string name="trace_location_organiser_warn_tan_error_ok_button">"OK"</string>
```
[the remark this answers is not in the document]
Comment (AlexanderAlferov) [reply Corona-Warn-App/src/main/res/values/event_registration_strings.xml:306]: This is header and OK button for dialog. Error texts has only body texts.

File: Corona-Warn-App/src/main/res/layout/trace_location_organizer_warn_tan_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        android:fillViewport="true"
+        tools:context=".ui.submission.tan.SubmissionTanFragment">
+
+        <include
+            android:id="@+id/tan_header"
+            layout="@layout/include_header"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:focusable="false"
+            app:icon="@{@drawable/ic_close}"
```
Comment (axelherbstreith) [inline Corona-Warn-App/src/main/res/layout/trace_location_organizer_warn_tan_fragment.xml:27]: should be ic_back according to figma

File: Corona-Warn-App/src/main/res/layout/trace_location_organizer_warn_tan_fragment.xml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            app:layout_constraintEnd_toEndOf="@id/guideline_end"
+            app:layout_constraintStart_toStartOf="@id/guideline_start" />
+
+        <Button
+            android:id="@+id/tan_button_enter"
+            style="@style/buttonPrimary"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:enabled="@{uiState.tanValid}"
+            android:text="@string/submission_tan_button_text"
```
Comment (axelherbstreith) [inline Corona-Warn-App/src/main/res/layout/trace_location_organizer_warn_tan_fragment.xml:60]: for me the button shows "Continue" (German: Weiter), but in Figma it says "send warning" (German: Warnung abschicken)

**Review by axelherbstreith — APPROVED** [review]
lgtm 👍

**Review by mtwalli — APPROVED** [review]
lgtm

**Pull request merged**